hazelcast/src/main/java/com/hazelcast/sql/impl/connector/SqlConnectorFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.connector;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public final class SqlConnectorFactory {
+
+    // TODO: replace it with connector class and its instantiation ???
+    // (TableSchemaField, Table, TableField, QueryDataType etc. need to be public then?)
+    private static final Map<String, SqlConnector> CONNECTORS_BY_TYPE = prepareConnectors();
+
+    private SqlConnectorFactory() {
+    }
+
+    private static Map<String, SqlConnector> prepareConnectors() {
```

Comment:
That is yet to be decided, we might pass the class name as `type` while creating the table so we could even accept user defined connectors, however that would require publishing certain classes that are private now - `Table`, `QueryDataType` etc.

Reply:
I see. Given that we do not expect more serious code conflicts around the optimizer code base for now, it may worth trying to add some real connectors (say, Kafka) to see how it to wire it up properly. Having such connectors not only will help us get a better understanding on how to wire them up but also will help us get rid of connectors for cluster-local maps. 
But maybe it is better to do this separately, to keep PRs more manageable.